modules/reindex/src/main/java/org/elasticsearch/index/reindex/DeleteByQueryRequest.java

```diff
@@ -76,4 +81,10 @@ public String toString() {
         searchToString(b);
         return b.toString();
     }
+
+    @Override
+    public List<? extends IndicesRequest> subRequests() {
+        assert getSearchRequest() != null;
+        return Collections.singletonList(getSearchRequest());
```

Comment:
that is a good point. why would it be a CompositeIndicesRequest to then have only one inner request? It should be search + delete but I guess it's tricky guess what it is going to delete depends on the query matches. Why does it need to be a CompositeIndicesRequest again?